tests/unit/cli/vyper_compile/test_compile_files.py

```diff
@@ -360,6 +360,104 @@ def test_archive_b64_output(input_files):
     assert out[contract_file] == out2[archive_path]
 
 
+def test_archive_compile_options(input_files):
+    tmpdir, _, _, contract_file = input_files
+    search_paths = [".", tmpdir]
+
+    options = ["abi_python", "json", "ast", "annotated_ast", "ir_json"]
+
+    for option in options:
+        out = compile_files([contract_file], ["archive_b64", option], paths=search_paths)
+
+        archive_b64 = out[contract_file].pop("archive_b64")
+
+        archive_path = Path("foo.zip.b64")
+        with archive_path.open("w") as f:
+            f.write(archive_b64)
+
+        # compare compiling the two input bundles
+        out2 = compile_files([archive_path], [option])
+
+        if option in ["ast", "annotated_ast"]:
+            # would have to normalize paths and imports, so just verify it compiles
+            continue
+
+        assert out[contract_file] == out2[archive_path]
+
+
+format_options = [
+    "bytecode",
+    "bytecode_runtime",
+    "blueprint_bytecode",
+    "abi",
+    "abi_python",
+    "source_map",
+    "source_map_runtime",
+    "method_identifiers",
+    "userdoc",
+    "devdoc",
+    "metadata",
+    "combined_json",
+    "layout",
+    "ast",
+    "annotated_ast",
+    "interface",
+    "external_interface",
+    "opcodes",
+    "opcodes_runtime",
+    "ir",
+    "ir_json",
+    "ir_runtime",
+    "asm",
+    "integrity",
+    "archive",
+    "solc_json",
+]
+
+
+def test_compile_vyz_with_options(input_files):
+    tmpdir, _, _, contract_file = input_files
+    search_paths = [".", tmpdir]
+
+    for option in format_options:
+        out_archive = compile_files([contract_file], ["archive"], paths=search_paths)
+
+        archive = out_archive[contract_file].pop("archive")
+
+        archive_path = Path("foo.zip.out.vyz")
+        with archive_path.open("wb") as f:
+            f.write(archive)
+
+        # compare compiling the two input bundles
+        out = compile_files([contract_file], [option], paths=search_paths)
+        out2 = compile_files([archive_path], [option])
+
+        if option in ["ast", "annotated_ast"]:
+            # would have to normalize paths and imports, so just verify it compiles
+            continue
+
+        if option in ["ir_runtime", "ir", "asm", "archive", "solc_json"]:
+            # TODO investigate why these don't pass the assert
```

Comment:
the should be correct, but would like to explore this before approving

Reply:
i suspect that has to do with codegen.reset_names(), which was fixed in 8f433f8de9ec3ead39e1691c45e2821fe8e3922b

Reply:
cf. as well: `vyper/ir/compile_ir.py:_next_symbol` may need to be reset.

Reply:
the assembly one should be fixed here: https://github.com/vyperlang/vyper/pull/4338/commits/e09f589607046cb16be9b09ff50ab984d8e79bc7

the others might need a bit more investigation. but i think the important thing here is that the bytecode output and integrity hashes are the same; if some metadata is different that is not such a big deal.

Reply:
debugged this a bit further and got the following:
```python
def _eq_helper(ir1, ir2):
    assert isinstance(ir1, IRnode) and isinstance(ir2, IRnode)
    if isinstance(ir1, IRnode):
        for n1, n2 in zip(ir1.args, ir2.args):
            if n1 != n2:
                if isinstance(n1, IRnode) and n1.value == "external_contract":
                    print("value comparison:", n1.value == n2.value)
                    print("args comparison:", n1.args == n2.args)
                    print("typ comparison:", n1.typ == n2.typ)
                    print("location comparison:", n1.location == n2.location)
                    print("mutable comparison:", n1.mutable == n2.mutable)
                    print("add_gas_estimate comparison:", n1.add_gas_estimate == n2.add_gas_estimate)
            if isinstance(n1, IRnode):
                _eq_helper(n1, n2

_eq_helper(out[contract_file]['ir_runtime'], out2[archive_path]['ir_runtime'])
```
output:
```
value comparison: True
args comparison: True
typ comparison: False
location comparison: True
mutable comparison: True
add_gas_estimate comparison: True
```

typ is `InterfaceT` where we don't have `def __eq__(self, other):` method but inherit from `_UserType` which defines:
```python
    def __eq__(self, other):
        return self is other
```